build.gradle

```diff
@@ -408,6 +417,87 @@ tasks.named("asciidoctor") {
       throw new IllegalArgumentException(errors.join('\n'))
     }
   }
+
+  doLast {
```

Comment:
:x: We should not add any more build logic into this file. I've been debating whether the docs need their own dedicated project or not. 
In any case, this logic is complex abd should be moved to a plugin in the `build-logic` module and have proper tests of their own.

Reply:
I moved the logic and added some tests